lib/Models/customComponents.js

```diff
@@ -0,0 +1,127 @@
+"use strict";
+
+/*global require*/
+var AssociativeArray = require('terriajs-cesium/Source/Core/AssociativeArray');
+var defined = require('terriajs-cesium/Source/Core/defined');
+
+var CustomComponentElement = require('./CustomComponentElement');
+
+var types = new AssociativeArray();
+var elements = [];
+
+/**
+ * A store of registered custom component types, eg. <chart>,
+ * and elements of custom components.
+ */
+var customComponents = {};
```

Comment:
Since this is a namespace (or maybe we could think of it as a static class), it should be PascalCased.